### PR TITLE
Handle RangeError from sequel: too many columns for content guessing

### PR DESCRIPTION
**Pull request by iriberri**
Although Postgres and therefore [OGR2OGR allow up to 1600](https://github.com/CartoDB/cartodb/blob/86c42a7306a8198d61ceb1fe6b7c3497e4ce3ce4/services/importer/lib/importer/ogr2ogr.rb#L96) columns in a dataset (depending on type), `sequel_pg` [sets the maximum number to 256](https://github.com/jeremyevans/sequel_pg/blob/79c768d9508c55f55db3f20331b9631b85ff97f0/ext/sequel_pg/sequel_pg.c#L16) in `SPG_MAX_FIELDS = 256`. (More info: https://github.com/CartoDB/cartodb/issues/3975)

The `table_sampler` file used in content guessing was crashing when getting a sample from the table whenever the dataset contained more than 256 columns. This is now rescued by raising a new exception `TooManyColumnsGuessedError` (other naming suggestions are welcome!).

With guessing disables, the old `TooManyColumnsError` would take care of the number of columns to be inserted in the dataset.

Fixes https://github.com/CartoDB/cartodb/issues/6263

### CODE REVIEW

**Comment by iriberri** [comment]
@csobier could you please review the wording at my changes in `import_error_codes.rb`? :) Thanks! You can also see the same text in the image I pasted in this PR description.

**Comment by iriberri** [comment]
cc @rafatower (or let me know if I should instead ping Mario to review)

File: lib/cartodb/import_error_codes.rb [hunk 1]
```diff
@@ -181,6 +181,11 @@ module CartoDB
       what_about: "The KML file you tried to import failed because a style element doesn't have an ID attribute. To fix this error, please open the file and add an ID to all the style tags.",
       source: ERROR_SOURCE_USER
     },
+    2010 => {
+      title: 'Too many columns for guessing',
+      what_about: "Data has too many columns to be guessed. The guessing process can only support up to 256 columns. Please, disable content guessing on import or reduce the number of columns in the file.",
```
Comment (csobier) [inline lib/cartodb/import_error_codes.rb]: Suggested edits:
Is it related to the data, or the file size? That will determine if the below should say "The Data" or "The file". Also, is "content guessing on import" an actual setting?

title: "The Data contains too many columns"
description: "The importer process can only upload up to 256 columns from your file. Reduce the number of columns in the file, or disable the content guessing on import."

Comment (iriberri) [inline lib/cartodb/import_error_codes.rb]: Notice that the "too many columns" (the one that refers to 1600) is already defined above.

This is just in the case when content guessing is enabled. If content guessing is not enabled, you can surely import more than 256 columns (I have for example a table with 270).

**Comment by rafatower** [comment]
I wouldn't say "for guessing" since sequel is part of our platform and it could fail at some other places. I.e: probably the editor won't work on such cases (but the SQL API will)

File: lib/cartodb/import_error_codes.rb (hunk 1 above)
Comment (csobier) [inline lib/cartodb/import_error_codes.rb]: Ah, so if it is specific to this option, perhaps this title would work?

title: There are too may columns for the Connect Dataset importer to guess
description: The importer process can only upload up to 256 columns from your file (when the "Let CartoDB automatically guess data types and content on import" checkbox is enabled." You can disable this option from the Connect Dataset dialog box, or reduce the number of columns in the file.

**Comment by iriberri** [comment]
@rafatower good point. I checked and (if I'm not wrong) the sampler is only being used right now for guessing (https://github.com/CartoDB/cartodb/blob/78922fb2e887ceda09f1acfd536ba2839d3ea08c/services/importer/lib/importer/content_guesser.rb#L152) although I totally agree that this should not be that specific.

What about a `TooManyColumnsRangeError` for the exception?

**Comment by rafatower** [comment]
My point is changing the message because `Sequel` is used in other places. Particularly in the `UserTable` and `Table` service. See https://github.com/CartoDB/cartodb/blob/master/app/models/table.rb#L816

So probably updating from the editor (which uses that function) will fail due to the `Sequel` library limitation and it is not something that will change in the near future.

**Comment by iriberri** [comment]
Paused until further limits discussions.

**Comment by iriberri** [comment]
image

**Comment by iriberri** [comment]
Finally tested in staging, @rafatower :-)

**Comment by rafatower** [comment]
:+1:

**Pull request merged**